tensorboard/plugins/graph/graph_util_test.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from google.protobuf import text_format
+import tensorflow as tf
+
+from tensorboard.compat.proto.graph_pb2 import GraphDef
+from tensorboard.plugins.graph import graph_util
+from tensorflow.python.platform import test
+
+
+class GraphUtilTest(tf.test.TestCase):
+  def test_combine_graph_defs(self):
+      expected_proto = '''
+        node {
+          name: "X"
+          op: "Input"
+        }
+        node {
+          name: "W"
+          op: "Input"
+        }
+        node {
+          name: "Y"
+          op: "MatMul"
+          input: "X"
+          input: "W"
+        }
+        node {
+          name: "X"
```

Comment:
For instance, I don’t understand this expected output. The node `X`
appears twice. Which node is the input to `Y`? which is the input to
`Q`? are the two `X`s actually the “same” node? if so, what happens if
one of them has `attr`s that the other doesn’t?